docs/Design-and-Modeling/10-api-security.md

```diff
@@ -0,0 +1,111 @@
+
+A security requirement, on the other hand, can exist both globally for the entire API or on an operation level. It matches the API as a whole or the individual API operation to one or more of the previously defined security schemes. Developers can use this to specify different levels of authorization explicitly, for example, by making read requests available without authentication, but requiring OAuth for write requests.
+
+## Creating Security Schemes
+
+![Add Security Scheme](../../assets/images/security-add-scheme.gif)
+
+You can define security schemes by following these steps:
+
+1. Open **API Overview** for your API.
+1. Click the **+** (plus) icon that appears next to _Security Schemes_.
+1. Select the scheme type in the leftmost dropdown menu.
+1. Give your scheme a name in the key field. This key must be unique within the API and will be used to reference the scheme in a security requirement.
+1. Specify other settings depending on the scheme type, as explained below.
+1. Optionally, click the description button (icon on the right) to add a human-readable description with more information about your scheme.
+
+### API Keys
+
+For API keys, select the _apiKey_ scheme type. An API key is a single token sent to the API server that the user has retrieved through an out-of-band mechanism. API keys can be a query parameter, a custom HTTP header, or a cookie.
+
+![API Key Configuration](../../assets/images/security-apikey.png)
+
+Use the right dropdown to toggle between _query_, _header_, and _cookie_ and use the name textbox to enter the query, cookie, or header name, respectively.
+
+When designing a new API with API Keys, we recommend the following:
+
+- Use _cookie_ when a web application uses your API in the browser, and you can use cookies for session management.
+- Use _query_ when your API is mostly exposing public data or doesn't have strict security constraints and may be used in environments where developers can only specify a URL but no custom headers.
+- In all other cases, use _header_ as the best practice that follows proper HTTP semantics.
+
+### HTTP Authentication
+
+For HTTP authentication, select the _http_ scheme type. HTTP authentication uses the standard HTTP _Authorization_ header defined in [RFC7235](https://tools.ietf.org/html/rfc7235).
+
+![HTTP Auth Configuration](../../assets/images/security-http.png)
+
+Use the right dropdown to switch between the different HTTP authentication schemes. OpenAPI and, by extension, Stoplight support all the [schemes registered at IANA](https://www.iana.org/assignments/http-authschemes/http-authschemes.xhtml). Further configuration is neither necessary nor possible.
+
+While OpenAPI supports the different variants of HTTP authentication, the majority of APIs use _basic_. With basic authentication, an API client sends the combination of a username and a password in the Authorization header. The method for retrieving the username and password is not part of the OpenAPI description.
```

Comment:
Maybe tell them to add the method to retrieve these credentials in a quickstart or auth documentation. We can do a small section about this "Writing documentation"?

Reply:
Added this section, please check.